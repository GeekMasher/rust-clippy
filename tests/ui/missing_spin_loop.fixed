// run-rustfix
#![warn(clippy::missing_spin_loop)]

use core::sync::atomic::{AtomicBool, Ordering};

fn main() {
    let b = AtomicBool::new(true);
    // This should lint
    while b.load(Ordering::Acquire) { std::hint::spin_loop() }

    // This is OK, as the body is not empty
    while b.load(Ordering::Acquire) {
        std::hint::spin_loop()
    }
    // TODO: also match on compare_exchange(_weak), but that could be
    // loop+match or while let
}
