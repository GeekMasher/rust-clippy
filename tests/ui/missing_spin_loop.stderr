error: busy-waiting loop should at least have a spin loop hint
  --> $DIR/missing_spin_loop.rs:9:37
   |
LL |     while b.load(Ordering::Acquire) {}
   |                                     ^^ help: try this: `{ std::hint::spin_loop() }`
   |
   = note: `-D clippy::missing-spin-loop` implied by `-D warnings`

error: aborting due to previous error

